torch/_refs/__init__.py

```diff
@@ -1067,6 +1072,34 @@ def amax(
     )
 
 
+@register_decomposition(torch.ops.aten.addr)
+@out_wrapper
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("self", "vec1", "vec2"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.NO_OPMATH,
+)
+def addr(
+    self: TensorLikeType, vec1: TensorLikeType, vec2: TensorLikeType,
+    beta: NumberType = 1,
+    alpha: NumberType = 1
+) -> TensorLikeType:
+    check(vec1.ndim == 1, lambda: f"addr: Expected 1-D argument vec1, but got {vec1.ndim}-D")
+    check(vec2.ndim == 1, lambda: f"addr: Expected 1-D argument vec2, but got {vec2.ndim}-D")
+    self = self.expand(vec1.shape[0], vec2.shape[0])
+    check(self.ndim == 2, lambda: f"2D tensor expected, got {self.ndim}D tensor for input")
+    check(self.shape[0] == vec1.shape[0] and self.shape[1] == vec2.shape[0],
+        lambda: f"size mismatch, input: {self.shape}, v1: {vec1.shape}, v2: {vec2.shape}")
+    if utils.is_boolean_dtype(self.dtype):
```

Comment:
should you also check `alpha` dtype (it should be of the same type kind as other inputs)?
@mruberry do you want to make a case again for `alpha` checking being done in the wrapper?

Reply:
Integers are OK though

```
>>> torch.addr(torch.tensor([[False]]), torch.tensor([False]), torch.tensor([True]), beta=1, alpha=2)
tensor([[False]])
```

Reply:
Extending the type promotion wrapper would probably make life easier

I think we should say that integer numbers can interact with bools and it's fine, however

Reply:
I tried the obvious thing which is 

```
@elementwise_type_promotion_wrapper(
    type_promoting_args=("self", "vec1", "vec2", "beta", "alpha"),
    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
)       
```

but this doesn't work as an integer alpha causes the entire tensor to promote to int64. I'm not sure what the recommend spelling of this change would be, it seems to have far reaching effects.